azure/scope/cluster.go

```diff
@@ -244,7 +244,7 @@ func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
 			Type:                 s.APIServerLB().Type,
 			SKU:                  infrav1.SKUStandard,
 			Role:                 infrav1.APIServerRole,
-			BackendPoolName:      s.APIServerLBPoolName(s.APIServerLB().Name),
+			BackendPoolName:      *s.APIServerLB().BackendPool.Name,
```

Comment:
we shouldn't dereference a pointer directly to avoid panics, let's use `to.String()` instead, or get rid of the pointer altogether (see my comment above)